library/lua/gui/widgets.lua

```diff
@@ -1708,9 +1708,9 @@ end
 local function get_button_token_base_pens(spec, x, y)
     local pen, pen_hover = COLOR_GRAY, COLOR_WHITE
     if spec.pens then
-        pen = type(spec.pens) == 'table' and spec.pens[y][x] or spec.pens
+        pen = type(spec.pens) == 'table' and #spec.pens > 0 and (#spec.pens[y] > 0 and spec.pens[y][x] or spec.pens[y]) or spec.pens
```

Comment:
```suggestion
        pen = type(spec.pens) == 'table' and safe_index(spec.pens, y, x) or spec.pens[y] or spec.pens
```

Reply:
to protect against the case where `spec.pens[y]` is `nil`

Reply:
Suggested change tested against all 3 mentioned test cases, all functioning withouth issues